### PR TITLE
Part of signed

### PR DESCRIPTION
**Pull request by HuiHuangShi**
Misra C required signed/unsigned conversion with cast.
Signed-off-by: Huihuang Shi <huihuang.shi@intel.com>
Acked-by: Eddie Dong <eddie.dong@intel.com>

### CODE REVIEW

**Comment by NanlinXie** [comment]
Can one of the admins verify this patch?

File: hypervisor/arch/x86/guest/vm.c [hunk 1]
```diff
@@ -98,15 +98,15 @@ int create_vm(struct vm_description *vm_desc, struct vm **rtn_vm)
 		goto err1;
 	}
 
-	for (id = 0; id < sizeof(long) * 8; id++)
-		if (bitmap_test_and_set(id, &vmid_bitmap) == 0)
+	for (id = 0U; id < (size_t)(sizeof(long) * 8U); id++)
+		if (bitmap_test_and_set(id, &vmid_bitmap) == false)
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vm.c]: This is already bool. so if (!bitmap_test_and_set(..))  should be fine.

File: hypervisor/arch/x86/guest/guest.c [hunk 2]
```diff
@@ -327,22 +327,24 @@ static inline int32_t _copy_gpa(struct vm *vm, void *h_ptr, uint64_t gpa,
...
 	do {
-		len = _copy_gpa(vm, h_ptr, gpa, size, 0, cp_from_vm);
-		if (len < 0)
-			return len;
+		ret = _copy_gpa(vm, h_ptr, gpa, size, 0, cp_from_vm);
+		if (ret < 0)
+			return ret;
 
+		len = (uint32_t) ret;
 		gpa += len;
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/guest.c]: gpa is uint32_t, we need type casting here.

File: hypervisor/arch/x86/guest/guest.c [hunk 3]
```diff
@@ -327,22 +327,24 @@ static inline int32_t _copy_gpa(struct vm *vm, void *h_ptr, uint64_t gpa,
...
 		h_ptr += len;
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/guest.c]: h_ptr is void *, is it ok for void * to add an integer?

File: hypervisor/arch/x86/guest/guest.c [hunk 4]
```diff
@@ -370,15 +373,16 @@ static inline int copy_gva(struct vcpu *vcpu, void *h_ptr, uint64_t gva,
...
 		}
 
-		len = ret = _copy_gpa(vcpu->vm, h_ptr, gpa, size,
+		ret = _copy_gpa(vcpu->vm, h_ptr, gpa, size,
 			PAGE_SIZE_4K, cp_from_vm);
 		if (ret < 0)
 			return ret;
 
+		len = (uint32_t) ret;
 		gva += len;
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/guest.c]: ditto

File: hypervisor/arch/x86/guest/guest.c [hunk 5]
```diff
@@ -370,15 +373,16 @@ static inline int copy_gva(struct vcpu *vcpu, void *h_ptr, uint64_t gva,
...
 		h_ptr += len;
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/guest.c]: ditto

File: hypervisor/debug/shell_internal.c [hunk 6]
```diff
@@ -802,14 +803,14 @@ int shell_vcpu_dumpmem(struct shell *p_shell,
...
-			for (i = 0; i < length/32; i++) {
+			for (i = 0U; i < length/32U; i++) {
 				snprintf(temp_str, MAX_STR_SIZE,
 					"=  0x%016llx  0x%016llx  0x%016llx  "
 					"0x%016llx\r\n", tmp[i*4], tmp[i*4+1],
 					tmp[i*4+2], tmp[i*4+3]);
 				shell_puts(p_shell, temp_str);
 			}
-			if (length > 32*(length/32)) {
+			if ((length % 32U) > 0) {
```
Comment (dongyaozu) [inline hypervisor/debug/shell_internal.c]: != 0  is better

File: hypervisor/common/hv_main.c [hunk 7]
```diff
@@ -56,7 +56,7 @@ void vcpu_thread(struct vcpu *vcpu)
 		}
 
 		vmexit_end = rdtsc();
-		if (vmexit_begin > 0)
+		if (vmexit_begin > 0UL)
```
Comment (dongyaozu) [inline hypervisor/common/hv_main.c]: != 0 is more clear

File: hypervisor/arch/x86/guest/guest.c (hunk 2 above)
Comment (HuiHuangShi) [reply hypervisor/arch/x86/guest/guest.c]: gpa is uint64_t, len is uint32_t, and len will be promoted implicitly.
 Add cast not look well.

File: hypervisor/arch/x86/guest/guest.c (hunk 3 above)
Comment (HuiHuangShi) [reply hypervisor/arch/x86/guest/guest.c]: It is ok to add an integer.

**Comment by HuiHuangShi** [comment]
related for loop or comma problems,another patch will come in,may be will rebase in the future.

**Pull request merged**